File: spec/factories/draft_page.rb
# typed: false
# frozen_string_literal: true

FactoryBot.define do
  factory :draft_page do
    space
    page
    editor { association :space_member }
    topic
    sequence(:title) { |n| "Page #{n}" }
    sequence(:body) { |n| "Body #{n}" }
    sequence(:body_html) { |n| "<div>Body #{n}</div>" }
    linked_page_ids { [] }
    modified_at { Time.current }
  end
end

File: go/internal/model/draft_page.go
package model

import (
	"time"
)

// DraftPage はページの下書きのドメインモデル
type DraftPage struct {
	ID                        DraftPageID
	SpaceID                   SpaceID
	PageID                    PageID
	SpaceMemberID             SpaceMemberID
	TopicID                   TopicID
	SuggestionPageID          *SuggestionPageID
	Title                     *string
	Body                      string
	BodyHTML                  string
	LinkedPageIDs             []PageID
	FeaturedImageAttachmentID *AttachmentID
	ModifiedAt                time.Time
	CreatedAt                 time.Time
	UpdatedAt                 time.Time

	Page  *Page
	Topic *Topic
}
